lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1312,10 +1312,15 @@ sub _failure_reason {
     my @failed_modules;
     my $modules = $self->modules;
     while (my $m = $modules->next) {
-        next if ($m->result eq INCOMPLETE);
-        push(@failed_modules, $m->name . ':' . $m->result);
+        if ($m->result eq FAILED) {
+            # in case we don't see an important module, we return all modules
+            push(@failed_modules, $m->name . ':' . $m->result);
+        }
+        if ($m->result eq FAILED && $m->fatal) {
+            return $m->name;
+        }
```

Comment:
The idea behind this special handling was that if you have one blocker bug you can tag it and it will be carried over, no matter how many modules before failed. But I don't think it's worth it - the system should be stable enough